src/index.ts

```diff
@@ -366,6 +373,10 @@ export default class StatsGatherer extends EventEmitter {
       remoteTracks: [],
     };
 
+    if (results.length === 0) {
```

Comment:
We still want to generate a 'disconnected' event from `handleConnectionStateChange` (but not update `lastResults`) and this seemed to be a simpler change to allow that to happen (see https://github.com/MyPureCloud/webrtc-stats-gatherer/pull/92/commits/a6145e4095a94b5f0ab5f0a8c48746dfc7a26524 for the other way I tried approaching this, though that change was broader-reaching that it strictly needed to be).

I'm open to other ideas, though.